### PR TITLE
Fix xalienfs scripts relocation and dylibs on MacOS

### PR DESCRIPTION
**Pull request by Atlantic777**
The xalienfs becomes dependency of AliEn-ROOT-Legacy and because of that the $GSHELL_ROOT environment variable should point to it as well.

### CODE REVIEW

File: alien-root-legacy.sh [hunk 1]
```diff
@@ -55,6 +58,7 @@ module load BASE/1.0 ${GCC_TOOLCHAIN_VERSION:+GCC-Toolchain/$GCC_TOOLCHAIN_VERSI
 
 # Our environment
 setenv ALIEN_ROOT_LEGACY_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
+setenv GSHELL_ROOT \$::env(ALIEN_ROOT_LEGACY_ROOT)
```
Comment (dberzano) [inline alien-root-legacy.sh]: Please add GSHELL_NO_GCC=1 in the Modulefile as well, thanks!

Comment (Atlantic777) [reply alien-root-legacy.sh]: Done.

**Comment by Atlantic777** [comment]
Back to WIP, the fix for .dylibs on MacOS will be added to this pull request.

File: alien-root-legacy.sh [hunk 2]
```diff
@@ -40,6 +43,11 @@ for RPKG in $BUILD_REQUIRES; do
   rm -f $INSTALLROOT/etc/modulefiles/{$RPKG,$RPKG.unrelocated} || true
 done
 
+if [[ $ARCHITECTURE == osx* ]]; then
+  ln -nfs $INSTALLROOT/lib/libAliEnROOTLegacy.dylib $INSTALLROOT/lib/libAliEnROOTLegacy.so
```
Comment (dberzano) [inline alien-root-legacy.sh]: The two symlinks here will not be relocated at install time, please use relative paths:

```bash
ln -nfs libBlahBlah.dylib $INSTALLROOT/lib/libBlahBlah.so
```

**Pull request merged**